controllers/update_test.go

```diff
@@ -1361,3 +1377,56 @@ func TestGetComponentName(t *testing.T) {
 	}
 
 }
+
+func TestSanitizeComponentName(t *testing.T) {
+
+	tests := []struct {
+		name          string
+		componentName string
+		want          string
+	}{
+		{
+			name:          "simple component name",
+			componentName: "devfile-sample-go-basic",
+			want:          "devfile-sample-go-basic",
+		},
+		{
+			name:          "simple component name, all numbers",
+			componentName: "123412341234",
+			want:          "comp-123412341234",
+		},
+		{
+			name:          "Empty string, should have a name generated for it",
+			componentName: "",
+		},
+		{
+			name:          "component name with uppercase",
+			componentName: "devfile-SAMPLE-gO-BASIC",
+			want:          "devfile-sample-go-basic",
+		},
+		{
+			name:          "component name with greater than 58 characters",
+			componentName: "devfile-sample-go-basic-devfile-sample-go-basic-devfile-sample",
+			want:          "devfile-sample-go-basic-devfile-sample-go-basic-devfile-sa-",
```

Comment:
Should we log a user friendly info message to say we are truncating the component name due to the 58 char limitation?